controllers/replace_failed_process_groups.go

```diff
@@ -58,6 +58,13 @@ func (c replaceFailedProcessGroups) reconcile(ctx context.Context, r *Foundation
 		}
 	}
 
+	// If the database is unavailable skip further steps as the operator is not able to make any good decision about the automatic
+	// replacements.
+	if !status.Client.DatabaseStatus.Available {
+		logger.Info("Skipping replaceFailedProcessGroups reconciler as the database is not available.")
+		return nil
```

Comment:
I am just curious, what is your decision logic for deciding whether this should requeue or not?

Reply:
the tradeoff being knowing what is coming next / that this is retried vs blocking other operations I suppose?

Reply:
In general we tried to build our subreconcilers in a way that if no errors occurs we try to proceed further, so doing a delayed requeue again makes sense to make sure the operator is retrying the reconciliation once the DB is available again. Could be an intermediate issue.